Add remote signing diagram (#167)
ietf-wg-scitt/draft-ietf-scitt-architecture: draft-ietf-scitt-architecture.md

When used in text, the corresponding terms are capitalized.
To ensure readability, only a core set of terms is included in this section.

> The terms "header", "payload", and "to-be-signed bytes" are defined in {{RFC9052}}.

Reviewer: :+1:

~~~
{: #example-transparent-statement-data-url align="left" title="Example Transparent Statement Data URL"}

# Signing Statements Remotely

Statements, such as digital artifacts or structured data regarding artifacts, can be too large or too sensitive to be send to a remote Transparency Services over the Internet.
In these cases a statement can also be hash, which becomes the payload included in COSE to-be-signed bytes.
A Signed Statement (cose-sign1) MUST be produced from the to-be-signed bytes according to {{Section 4.4 of RFC9052}}.

~~~aasvg

 .----+-----.
|  Artifact  |
 '----------'
    |   |
    v   |               .
 .--+-------.          / \
| Statement  +--+     /   \
 '----------'   |    /     \     .----------.
        |       +-->+  OR   +-->+  Payload   |
        v       |    \     /     '---+------'
 .------+---.   |     \   /          |
|    Hash    +--+      \ /           |
 '----+-----'           '            |
                                     |

           ...  Producer Network ...

                      ...

           ...   Issuer Network ...

                                     |
+-----------+                        |

> | Identity  |     (iss, x5t)         |

Reviewer 2: As a "document", should this have rounded corners?
I have time this afternoon to reivew and make PRs to tweak before approving.

Author: This box represents a concept for an identity document, not a concrete identity document such as a certificate.

I don't intend to update the PR, but you are welcome to.

Reviewer: As the Identity Document pre-exists outside of that data flow, I'd be okay with keeping that box different.

| Document  +--------------------+   |
+-----+-----+                    |   |
      ^                          |   |
      |                          |   |
 .----+-------.                  |   |
| Private Key  |                 |   |
 '----+-------'                  v   |
      |                     .----+-----.

>       |                    |   Header   |

Reviewer 3: The Header part and the "To be Signed bytes are mis-leading to me"

Reviewer 3: Specially the `Payload` sections `Arrow` enters into the Header section. What are we trying to depict?

Reviewer 3: Also if I am not mis-taken `To be signed bytes` in our case is actually the `payload` ?

If it is, then can we keep the two same so as to logically link the two sections of the diagrams without physically linking them?

      |                     '----+-----'
      |                          |   |
      v                          v   v
    .-+-----------.     .--------+---+---.
   /             /     /                  \
  /    Sign     +<----+ To Be Signed Bytes |
 /             /       \                  /

> '-------------'         '----------------'

Reviewer 2: ```suggestion
'-----+-------'         '----------------'
```

Reviewer 2: Minor fix to differentiate lines "behind", vs. from. 
In this case, I believe @OR13 is showing the COSE Sign 1 Document comes from the Sign process.

Author: The lines are a correct as they are.

data:application/cose;base64,SGVsb...xkIQ==
~~~
{: #example-transparent-statement-data-url align="left" title="Example Transparent Statement Data URL"}

# Signing Statements Remotely

> Some statements are too large, or sensitive to send over a network to a remote signer.

Reviewer: ```suggestion
Some statements are too large or too sensitive to send over a network to a remote signer.
```

> Some statements are too large, or sensitive to send over a network to a remote signer.
> 
> Recall that a Statement is an artifact, structured data regarding an artifact, or a hash.
> 
> A Statement becomes a payload, which becomes to-be-signed bytes.
> 
> A Signed Statement (cose-sign1) MUST be produced from the to-be-signed bytes according to {{Section 4.4 of RFC9052}}.

Reviewer: ```suggestion
Some statements, such as digital artifacts structured data regarding an artifacts, are too large or too sensitive to be send to a remote Transparency Services over the Internet.
In these cases a statement can also be hash, which becomes the payload included in COSE to-be-signed bytes.
A Signed Statement (cose-sign1) MUST be produced from the to-be-signed bytes according to {{Section 4.4 of RFC9052}}.
```

Reviewer: Wait, how is this PR related to #155?
